ext/SummationByPartsOperatorsOptimForwardDiffExt.jl

```diff
@@ -0,0 +1,222 @@
+module SummationByPartsOperatorsOptimForwardDiffExt
+
+using Optim: Optim, Options, LBFGS, optimize, minimizer
+using ForwardDiff: ForwardDiff
+
+using SummationByPartsOperators: SummationByPartsOperators, GlaubitzNordströmÖffner2023, MatrixDerivativeOperator
+using LinearAlgebra: Diagonal, LowerTriangular, dot, diag, norm, mul!
+using PreallocationTools: DiffCache, get_tmp
+using SparseArrays: spzeros
+
+function SummationByPartsOperators.function_space_operator(basis_functions,
+                                                           x_min::T, x_max::T, nodes::Vector{T},
+                                                           source::SourceOfCoefficients;
+                                                           derivative_order = 1, accuracy_order = 0,
+                                                           options = Options(g_tol = 1e-14, iterations = 10000)) where {T, SourceOfCoefficients}
+
+    @assert derivative_order == 1 "Only first derivative operators are supported"
+    sort!(nodes)
```

Comment:
I'm not sure if sorting the nodes is reasonable or necessary. It also works without sorting, but having a non-sorted grid in the `MatrixDerivativeOperator` feels a bit strange to me. What do you think, @ranocha?

Reply:
I prefer sorting the nodes for consistency with the other operators 👍